app/code/Magento/Catalog/Controller/Adminhtml/Category/Add.php

```diff
@@ -61,9 +68,8 @@ public function execute()
             $category->addData($categoryData);
         }
 
-        $resultPageFactory = $this->_objectManager->get(\Magento\Framework\View\Result\PageFactory::class);
-        /** @var \Magento\Backend\Model\View\Result\Page $resultPage */
-        $resultPage = $resultPageFactory->create();
+        /** @var Page $resultPage */
+        $resultPage = $this->resultFactory->create(ResultFactory::TYPE_PAGE);
```

Comment:
Will check if such change is needed together with other PR.

Reply:
It's needed because `$this->resultFactory` we have in all admin controllers, it comes from `Magento\Backend\App\Action\Context` in constructor. It's better to have one dependency, that already was there, rather than introducing new and new factory dependencies for all types of results.

Reply:
@ihor-sviziev I'm more concerned regarding `$this->resultRedirectFactory` not used.

I'm ok with this OM replacement.